device/thunder_ripple_sdk/src/processors/thunder_device_info.rs

```diff
@@ -721,22 +717,17 @@ impl ThunderDeviceInfoRequestProcessor {
 
     async fn get_hdcp_status(state: &CachedState) -> HDCPStatus {
         let mut response: HDCPStatus = HDCPStatus::default();
-        match state.get_hdcp_status() {
-            Some(status) => response = status,
-            None => {
-                let resp = state
-                    .get_thunder_client()
-                    .call(DeviceCallRequest {
-                        method: ThunderPlugin::Hdcp.method("getHDCPStatus"),
-                        params: None,
-                    })
-                    .await;
-                info!("{}", resp.message);
-                if let Ok(thdcp) = serde_json::from_value::<ThunderHDCPStatus>(resp.message) {
-                    response = thdcp.hdcp_status;
-                    state.update_hdcp_status(response.clone());
-                }
-            }
+        let resp = state
+            .get_thunder_client()
+            .call(DeviceCallRequest {
+                method: ThunderPlugin::Hdcp.method("getHDCPStatus"),
+                params: None,
+            })
+            .await;
+        info!("{}", resp.message);
+        if let Ok(thdcp) = serde_json::from_value::<ThunderHDCPStatus>(resp.message) {
+            response = thdcp.hdcp_status;
+            state.update_hdcp_status(response.clone());
```

Comment:
If `hdcp_status` is not cacheable then remove that from `CachedDeviceInfo`.

Reply:
I also had a related question @nnaveen979, is ripple currently polling for HDCP status? If so, where?

Reply:
Discussed resolution is that we will move forward with this solution for now and I will create a new ticket to fix/improve HDP/HDCP event handling/caching/etc.